index.js

```diff
@@ -48,7 +48,11 @@ function parseLink(link) {
 
 module.exports = function (linkHeader) {
   if (!linkHeader) return null;
-
+  const MAX_LINKHEADER_LENGTH = 2000;
+  if (linkHeader.length > MAX_LINKHEADER_LENGTH) {
+    throw new Error('Input string too long, it should be under ' + MAX_LINKHEADER_LENGTH + ' characters.');
+  }
```

Comment:
We need to make this configurable with an env var, otherwise people's servers will start crashing when this code hits.
Could be `PARSE_LINK_HEADER_THROW_ON_MAXLEN_EXCEEDED` and if that is not found we could just fail silently.